src/main/java/org/fenixedu/academic/domain/student/personaldata/ProfessionCategoryType.java

```diff
@@ -0,0 +1,48 @@
+package org.fenixedu.academic.domain.student.personaldata;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.fenixedu.academic.domain.exceptions.DomainException;
+import org.fenixedu.academic.util.Bundle;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.bennu.core.i18n.BundleUtil;
+import org.fenixedu.commons.i18n.LocalizedString;
+
+public class ProfessionCategoryType extends ProfessionCategoryType_Base {
+
+    protected ProfessionCategoryType() {
+        super();
+        setRootDomainObject(Bennu.getInstance());
+    }
+
+    public static ProfessionCategoryType create(String code, LocalizedString name, boolean active) {
+        if (findByCode(code).isPresent()) {
+            throw new DomainException(
+                    BundleUtil.getString(Bundle.APPLICATION, "error.ProfessionCategoryType.code.already.exists", code));
+        }
+
+        ProfessionCategoryType professionCategoryType = new ProfessionCategoryType();
+        professionCategoryType.setCode(code);
+        professionCategoryType.setName(name);
+        professionCategoryType.setActive(active);
+        return professionCategoryType;
+    }
+
+    public void delete() {
+        setRootDomainObject(null);
+        super.deleteDomainObject();
+    }
+
```

Comment:
The `setCode(code)` method should be overriden in order to check if already exists another instance with the same code